torchtext/datasets/text_classification.py

```diff
@@ -113,33 +122,61 @@ def get_vocab(self):
         return self._vocab
 
 
-def _setup_datasets(dataset_name, root='.data', ngrams=2, vocab=None, include_unk=False):
+def _setup_datasets(dataset_name, root='.data', ngrams=2, vocab=None,
+                    removed_tokens=[], tokenizer=None,
+                    data_select=('train', 'test')):
+    if not tokenizer:
+        tokenizer = get_tokenizer("basic_english")
+
+    if isinstance(data_select, str):
+        data_select = [data_select]
+    if not set(data_select).issubset(set(('train', 'test'))):
+        raise TypeError('data_select is not supported!')
+
     dataset_tar = download_from_url(URLS[dataset_name], root=root)
     extracted_files = extract_archive(dataset_tar)
 
+    path = {}
     for fname in extracted_files:
-        if fname.endswith('train.csv'):
-            train_csv_path = fname
-        if fname.endswith('test.csv'):
-            test_csv_path = fname
+        if fname.endswith('train.csv') and 'train' in data_select:
+            path['train'] = fname
+        if fname.endswith('test.csv') and 'test' in data_select:
+            path['test'] = fname
 
     if vocab is None:
-        logging.info('Building Vocab based on {}'.format(train_csv_path))
-        vocab = build_vocab_from_iterator(_csv_iterator(train_csv_path, ngrams))
+        if 'train' not in data_select:
+            raise TypeError("Must pass a vocab if train is not selected.")
+        logging.info('Building Vocab based on train data')
+        if dataset_name == 'IMDB':
```

Comment:
I think that'll also be a great example of writing preprocessing on a generic source where the data might be backed by multiple or a single file in various formats.

Reply:
Yes. I think I need to write a demo to show how to build a pipeline for torchtext data processing, in case that users would like to load their own data.

Reply:
On that note, can you also add to this PR an example code on how to load `train1` in the example you already provide?

Reply:
Sure. Added it.